app/src/lib/get-main-guid.ts

```diff
@@ -0,0 +1,45 @@
+import { app } from 'electron'
+import { readFile, writeFile } from 'fs/promises'
+import { join } from 'path'
+import { uuid } from './uuid'
+
+let cachedGUID: string | null = null
+
+/** Get the GUID for the Main process. */
+export async function getMainGUID(): Promise<string> {
+  if (!cachedGUID) {
+    cachedGUID = await readGUIDFile()
+  }
+
+  return cachedGUID
+}
```

Comment:
There's two approaches we can take here to clean this up a little. Either we leverage the new `??=` opertor

```ts
let cachedGUID: string | null = null

/** Get the GUID for the Main process. */
export const getMainGUID = async () => (cachedGUID ??= await readGUIDFile())
```

I did note, however that our version of prettier doesn't understand logical assignment operators. I'll fix that separately. We could also leverage memoizeOne in which case we don't need the cachedGUID at all.

```suggestion
import memoizeOne from 'memoize-one'

/** Get the GUID for the Main process. */
export const getMainGUID = memoizeOne(readGUIDFile)
```

The `memoizeOne` method will cache errors as well though whereas the previous method will keep retrying `readGUIDFile()` on each call but it looks like `readGUIDFile` isn't intended to throw ever?

Reply:
I love that `memoizeOne` approach! As you say, `readGUIDFile` isn't supposed to throw errors (happy change that if you have a good reason for it, of course).